### PR TITLE
[Feature][API][backend] Query all upstream and downstream dependencies of the task

### PR DESCRIPTION
**Pull request by IamMujuziMoses**
## What is the purpose of the pull request

- [x] Upstream Dependency
- [x] Downstream Dependency
- [ ] Upstream and Downstream dependency

## Brief change log

*ProcessDefinitionController.java*
*ProcessDefinitionService.java*
*ProcessDefinitionServiceImpl.java*
  - *Added method `queryUpstreamTaskDependencies`*
  - *Added method `queryDownstreamTaskDependencies`*

*ProcessDefinitionControllerTest.java*
*ProcessDefinitionServiceTest.java*
 - *Added method `testQueryUpstreamTaskDependencies`*
 - *Added method `testQueryDownstreamTaskDependencies`*

*DependentParameters.java*
*DependentItem.java*
*DependentTaskModel.java*
 - *Added `toString()` method*

*Status.java*
 - *Added `QUERY_TASK_DEPENDENCIES_BY_TASK_NAME_ERROR`*

### CODE REVIEW

File: dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/service/impl/ProcessDefinitionServiceImpl.java [hunk 1]
```diff
@@ -1797,5 +1800,94 @@ private void checkBatchOperateResult(String srcProjectName, String targetProject
...
+        List<TaskNode> taskNodeList = (processData.getTasks() == null) ? new ArrayList<>() : processData.getTasks();
+
+        for (TaskNode taskNode : taskNodeList) {
+            if (taskNode.getName().equals(taskName)) {
+                //getPreTasks() holds inner dependency information
+                depList.add(taskNode.getPreTasks());
+            }
+
+        }
+
```
Comment (lgcareer) [inline dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/service/impl/ProcessDefinitionServiceImpl.java:1842]: Hi,if the pretasks contains dependence task,you should parse the dependene json.

Comment (lgcareer) [reply dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/service/impl/ProcessDefinitionServiceImpl.java:1842]: You can also abstract a class to describe the dependent item,it need includes common attributes：project id,process definition id,task name

**Comment by IamMujuziMoses** [comment]
@lgcareer I have been parsing the `dependence` field to get the downstream dependencies but I have failed to locate `shell_02` or the equivalent information from the dependence field, where is that information stored?!
<sub>[View in Slack](https://the-asf.slack.com/archives/G01L7J4ST2Q/p1614339260004900)</sub>

**Comment by IamMujuziMoses** [comment]
Hi @lgcareer @dailidong sonar is reporting some code smell and vulnerability but for the code smell, that is the needed return type and for the vulnerability, we need to log the user-controlled data as protocol, so that is a mistake in the sonar results

**Comment by kezhenxu94** [comment]
It's been 1+ year and too many conflicts, let me close this and @IamMujuziMoses if you are interesting in continue please feel free to rebase the branch and open a new PR

**Pull request closed without merging**